From c56e9c197a6e6ad0f7a132d1337c60171f66b6ce Mon Sep 17 00:00:00 2001
From: tedyu <yuzhihong@gmail.com>
Date: Mon, 20 Apr 2015 14:27:47 -0700
Subject: [PATCH 195/224] HBASE-13514 Fix test failures in
 TestScannerHeartbeatMessages caused by incorrect
 setting of hbase.rpc.timeout (Jonathan Lawlor)

Reason: Bug
Author: Jonathan Lawlor
Ref: CDH-27345
---
 .../regionserver/TestScannerHeartbeatMessages.java |    5 +++--
 1 file changed, 3 insertions(+), 2 deletions(-)

diff --git a/hbase-server/src/test/java/org/apache/hadoop/hbase/regionserver/TestScannerHeartbeatMessages.java b/hbase-server/src/test/java/org/apache/hadoop/hbase/regionserver/TestScannerHeartbeatMessages.java
index f86c178..76fb0be 100644
--- a/hbase-server/src/test/java/org/apache/hadoop/hbase/regionserver/TestScannerHeartbeatMessages.java
+++ b/hbase-server/src/test/java/org/apache/hadoop/hbase/regionserver/TestScannerHeartbeatMessages.java
@@ -104,7 +104,7 @@ public class TestScannerHeartbeatMessages {
   // Time, in milliseconds, that the client will wait for a response from the server before timing
   // out. This value is used server side to determine when it is necessary to send a heartbeat
   // message to the client
-  private static int CLIENT_TIMEOUT = 500;
+  private static int CLIENT_TIMEOUT = 2000;
 
   // The server limits itself to running for half of the CLIENT_TIMEOUT value.
   private static int SERVER_TIME_LIMIT = CLIENT_TIMEOUT / 2;
@@ -113,7 +113,7 @@ public class TestScannerHeartbeatMessages {
   private static int DEFAULT_ROW_SLEEP_TIME = SERVER_TIME_LIMIT / 2;
   // By default, at most cells for two column families are retrieved before the time limit is
   // reached
-  private static int DEFAULT_CF_SLEEP_TIME = SERVER_TIME_LIMIT / 2;
+  private static int DEFAULT_CF_SLEEP_TIME = DEFAULT_ROW_SLEEP_TIME / NUM_FAMILIES;
 
   @BeforeClass
   public static void setUpBeforeClass() throws Exception {
@@ -173,6 +173,7 @@ public class TestScannerHeartbeatMessages {
 
   @AfterClass
   public static void tearDownAfterClass() throws Exception {
+    TEST_UTIL.deleteTable(TABLE_NAME);
     TEST_UTIL.shutdownMiniCluster();
   }
 
-- 
1.7.9.5

